pkg/systemd/generate/common.go

```diff
@@ -111,6 +111,24 @@ func escapeSystemdArg(arg string) string {
 	return arg
 }
 
+func removeSdNotifyArg(args []string, argCount int) []string {
+	processed := make([]string, 0, len(args))
+	for i := 0; i < len(args)-argCount; i++ {
+		s := args[i]
+
+		switch {
+		case s == "--sdnotify":
+			i++
+			continue
+		case strings.HasPrefix(s, "--sdnotify="):
+			continue
+		}
+		processed = append(processed, s)
+	}
+	processed = append(processed, args[len(args)-argCount:]...)
+	return processed
+}
+
```

Comment:
I guess command-line option values can't have the value `--sdnotify`?
(That could confuse the parsing)

Reply:
This is not possible, the option always needs an argument.

Reply:
I was thinking about `--some-other-option value` where value is `--sdnotify`.

Maybe it's not possible. In any case, it's too unrealistic.

Reply:
Yeah that could be a problem. However the other flags are handled in the same way. I think we can ignore this for now until somebody comes up with a real problem. Note that we will never touch the arguments after the image since they are for the container process.